features/step_definitions/run_steps.rb

```diff
@@ -1,40 +1,37 @@
-When(/^(I|my coworker) runs? `([^`]+)`( while allowing errors)?$/) do |who, commands, allow_failures|
+When(/^(I|my coworker) runs? `([^`]+)`$/) do |who, commands|
   user = (who == 'I') ? :developer : :coworker
   in_repository user do
-    commands.split(';').each do |command|
-      run command.strip, allow_failures: allow_failures
-    end
+    commands.split(';').each { |command| run command.strip }
   end
 end
 
 
-When(/^I run `([^`]+)` with the last( two)? commit shas?( while allowing errors)?$/) do |command, two, allow_failures|
+When(/^I run `([^`]+)` with the last( two)? commit shas?$/) do |command, two|
   count = two ? 2 : 1
   shas = recent_commit_shas(count).join(' ')
-  run "#{command} #{shas}", allow_failures: allow_failures
+  step "I run `#{command} #{shas}`"
 end
 
 
-When(/^I run `(.+?)` and enter (.+?)$/) do |command, input|
-  inputs = prepare_user_input input
-  @result = run command, inputs: inputs, allow_failures: true
+When(/^I run `(.+?)` and enter "(.+?)"$/) do |command, input|
+  @result = run command, input: input
 end
 
 
+When(/^I run `(.+?)` and enter an empty commit message$/) do |command|
+  step "I run `#{command}` and enter \"dGZZ\""
```

Comment:
Maybe just a quick comment explaining what `dGZZ` does. I had no idea until you told me (always exited vim with `:wq`)